Replace `logging` with `loguru`

lochhh opened the pull request:
Before submitting a pull request (PR), please read the [contributing guide](https://github.com/neuroinformatics-unit/.github/blob/main/CONTRIBUTING.md).

Please fill out as much of this template as you can, but if you have any problems or questions, just leave a comment and we will help out :)

## Description

**What is this PR**

- [ ] Bug fix
- [ ] Addition of a new feature
- [ ] Other

**Why is this PR needed?**

**What does this PR do?**

## References

Please reference any existing issues/PRs that relate to this PR.

## How has this PR been tested?

Please explain how any new code has been tested, and how you have ensured that no existing functionality has changed.

## Is this a breaking change?

If this PR breaks any existing functionality, please explain how and why.

## Does this PR require an update to the documentation?

If any features have changed, or have been added. Please explain how the
documentation has been updated.

## Checklist:

- [ ] The code has been tested locally
- [ ] Tests have been added to cover all new functionality
- [ ] The documentation has been updated to reflect any changes
- [ ] The code has been formatted with [pre-commit](https://pre-commit.com/)

niksirbi commented on tests/conftest.py:
```diff
@@ -21,24 +21,32 @@ def _to_module_string(path: str) -> str:
-        logger_name="movement",
-        log_directory=(tmp_path / ".movement"),
+@pytest.fixture
+def caplog(caplog: LogCaptureFixture):
```
I'm not sure I fully understand what this override does, but it seems to allow us stick to all of our existing uses of `caplog` in the tests, so I'm happy.

lochhh replied:
Added more details to clarify
https://github.com/neuroinformatics-unit/movement/blob/deb55013d2a4162970f0b5cd6ea55e3875f59308/tests/conftest.py#L42-L45

lochhh commented on movement/io/save_poses.py:
```diff
@@ -412,7 +410,7 @@ def _validate_file_path(
         )
     except (OSError, ValueError) as error:
         logger.error(error)
-        raise error
+        raise
```
Yeah that works too, but we'd hit [raise-without-from-inside-except (B904)](https://docs.astral.sh/ruff/rules/raise-without-from-inside-except/#raise-without-from-inside-except-b904), which suggests us to use `raise logger.error(error) from error`, which in this case is redundant since we're raising the original exception. 
So our options are:
- two lines (as it is now)
- single line with `# noqa: B904`

niksirbi replied:
Ah I see, so B904 was the reason. I honestly don't care here, feel free to choose whichever option you prefer.

lochhh commented on movement/utils/logging.py:
```diff
@@ -1,112 +1,84 @@
+
+        This method configures a rotating log file that
+        logs at the DEBUG level with a maximum size of 5 MB
+        and retains the last 5 log files.
```
I've now changed this to retain only 1 file in addition to the current file

lochhh commented:
> Thanks @lochhh!
> 
> I tried using `movement` through Python console and jupyter notebooks, while also keeping an eye on the log file, and this PR seems to work as advertised. Also kudos for maintaining a somewhat backwards-consistent way for us to raise errors and warnings (the syntax is very similar to what we had before).
> 
> I think this is good to go, barring a few not-so-important comments I've left. Also make sure to rebase to `main` so the changes in logging also apply to recently added stuff. One case which might be tricky is the deprecation warning I added in #455 (in `filtering.py`). Ideally we want to use the new syntax for it but also ensure that specifically a `DeprecationWarning` is raised.

Thanks @niksirbi. `warnings.warn()` messages are now redirected to the logger. I think it's worth keeping both logger.warning() and warnings.warn() for different purposes, so I've replaced some `print` and `logger.warning` calls with `warnings.warn` (see updated [PR description](https://github.com/neuroinformatics-unit/movement/pull/467#issue-2895141315) sections: Integration with `warnings` module, `print` vs `warnings` vs `logger`).

lochhh commented:
> > Thanks @niksirbi. `warnings.warn()` messages are now redirected to the logger. I think it's worth keeping both logger.warning() and warnings.warn() for different purposes, so I've replaced some `print` and `logger.warning` calls with `warnings.warn` (see updated [PR description](https://github.com/neuroinformatics-unit/movement/pull/467#issue-2895141315) sections: Integration with `warnings` module, `print` vs `warnings` vs `logger`).
> 
> Thanks @lochhh. I think you did a great job with this. I benefitted a lot from reading your updated PR description, because when to use logger/warnings/print was not entirely clear in my mind. I have only one final request to make: could you copy some of that information over to the contributing guide? It would be good to have a canonical place to point contributors to, when we inevitably have to explain this again.

Added a section describing this in Development guidelines
